yaml-tests/src/test/resources/valid-identifiers.yamsql

```diff
@@ -262,6 +270,83 @@ test_block:
     -
       - query: select count(*) from "TEMPLATES" where template_name = 'टेम्पलेट'
       - result: [{1}]
+---
+setup:
+  connect: "jdbc:embed:/__SYS?schema=CATALOG"
+  steps:
+    - query: drop schema template if exists test_template_with_inavlid_identifiers
+---
+test_block:
+  connect: "jdbc:embed:/__SYS?schema=CATALOG"
+  tests:
+    -
+      # invalid table name
+      - query: create schema template test_template_with_inavlid_identifiers
+               create table "__$yay"(id2 bigint, col6 s2 ARRAY, primary key(id2))
```

Comment:
It feels like this should be legal. I think I'd propose an alternative approach: we skip the first two underscores, but we still translate the rest of the string. We need to also ban identifiers that start with `$` and `.` to avoid ambiguities (for example, `__1f` and `$f` would both become the protobuf identifier `__1f` if all we did was skip translation of `__`), but that seems like something we can accommodate.

Furthermore, it feels like we could use a test that starts with a protobuf schema definition where there identifiers beginning with `__`, and then we validate that we can properly access them.

Reply:
Related to the issue of ambiguity with`__1f` and `$f` is the fact we would also have to ban `__x` prefixed identifiers  , where x is a digit. Because, `__1f` is ambigous is reverse translation

Reply:
Reduced the scope of restriction to only explicitly apply to `__0`, `__1` and `__2`